decoders/connector/lansitec/compact-bluetooth-gateway/v1.0.0/payload.js

```diff
@@ -1,501 +1,288 @@
-// Decode decodes an array of bytes into an object.
-//  - fPort contains the LoRaWAN fPort number
-//  - bytes is an array of bytes, e.g. [225, 230, 255, 0]
-//  - variables contains the device variables e.g. {"calibration": "3.5"} (both the key / value are of type string)
-// The function must return an object, e.g. {"temperature": 22.5}
-// gateway上行数据解析代码
-function Decode(fPort, bytes) {
-  // 根据消息开头的第一个字节解析数据           
-  var type = ((bytes[0] >> 4) & 0x0F);
-  if(type == 0x01){
-    var beaconBytes = bytes.slice(2, 5);
-    // console.log(beaconBytes);
-    // 校验过
-    return registerUplinkMessage("", bytes);
-  }           
-  if (type == 0x02){          
-    // 校验过
-    return heartbeatUplinkMessage("", bytes);
-  }  
-  if (type == 0x06) {
-    // 校验过
-    return alarmUplinkMessage("", bytes);
-  }            
-  if (type == 0x08) {
-    // 校验过
-    return beaconUplinkMessage("", bytes);
-  } 
-  if (type == 0x09) {
-    // 校验
-    return braceletUplinkMessage("", bytes);
-  } 
-  if (type == 0x0A) {
-    // 校验过 - 格式有待完善
-    return lightPerceptionUplinkMessage("", bytes);
-  } 
-  if (type == 0x0B) {
-    // 校验过 - 格式有待完善
-    return gSensorUplinkMessage("", bytes);
-  } 
-  if (type == 0x0D) {
-    // 校验
-    return beaconListUplinkMessage("", bytes);
-  } 
-  if (type == 0x0F) {
-    // 校验过
-    return acknowledgeUplinkMessage("", bytes)
-  } 
-  
-}
-// 4.4.1 Register, 0x01设备注册消息
-function registerUplinkMessage(prefix, bytes) {
-  var data = {}; 
-  data[prefix+"type"] = "Register";
-  data[prefix+"adr"] = ((bytes[0] >> 3) & 0x01) === 0? "OFF" : "ON";
-  data[prefix+"mode"] = (bytes[0] & 0x07);
-  switch (data[prefix+"mode"]) {
+// BluetoothGateway

```

Comment:
This decoder must have a unit test, you can use the example I posted in the comments above, but if you can add the unit test for the payload parser it will be great.